### PR TITLE
fix: validation schema with `schema-utils`

### PR DESCRIPTION
**Pull request by evilebottnawi**
Fixes https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/524

We can avoid using `$schema` as we can do in `webpack` (https://github.com/webpack/webpack/tree/master/schemas)

### CODE REVIEW

**Comment by d3viant0ne** [comment]
@bebraw - Wouldn't it just make more sense to migrate this to schema utils and maintain the `ajv` dependency there?

**Comment by bebraw** [comment]
@d3viant0ne Agreed.

@evilebottnawi Can you make that change? Better solution over longer term.

**Comment by evilebottnawi** [comment]
@bebraw let's do it

File: index.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ var ExtractedModule = require("./ExtractedModule");
 var Chunk = require("webpack/lib/Chunk");
 var OrderUndefinedError = require("./OrderUndefinedError");
 var loaderUtils = require("loader-utils");
-var schemaTester = require('./schema/validator');
+var validateOptions = require('schema-utils');
 var loaderSchema = require('./schema/loader-schema');
```
Comment (michael-ciniawsky) [inline index.js]: The two schemas can be removed, just pass the `path/to/schema.json` to `validateOptions`

File: index.js [hunk 2]
```diff
@@ -122,7 +122,7 @@ function ExtractTextPlugin(options) {
 	if(isString(options)) {
 		options = { filename: options };
 	} else {
-		schemaTester(pluginSchema, options);
+		validateOptions(pluginSchema, options, ' Extract Text Plugin');
```
Comment (michael-ciniawsky) [inline index.js]: `./schema/plugin.json` && ` Extract Text Plugin` => `Extract Text Plugin` (no leading space) 

Please rename `plugin-schema.js` => `plugin.json`

File: index.js [hunk 3]
```diff
@@ -201,7 +201,7 @@ ExtractTextPlugin.prototype.extract = function(options) {
 	if(Array.isArray(options) || isString(options) || typeof options.options === "object" || typeof options.query === 'object') {
 		options = { loader: options };
 	} else {
-		schemaTester(loaderSchema, options);
+		validateOptions(loaderSchema, options, 'Extract Text Loader');
```
Comment (michael-ciniawsky) [inline index.js]: `./schema/loader.json` && ` Extract Text Loader` => `Extract Text Plugin (Loader)`

Please rename `loader-schema.js` => `loader.json`

File: test/extract.test.js [hunk 4]
```diff
@@ -29,7 +29,10 @@ describe("ExtractTextPlugin.extract()", function() {
 					ExtractTextPlugin.extract({style: 'file.css'});
 				},
 				function(err) {
-					return err.message === 'data[\'style\'] should NOT have additional properties';
+					console.log(err.message)
```
Comment (michael-ciniawsky) [inline test/extract.test.js]: Needed for testing or forgotten to remove ? 😛

Comment (evilebottnawi) [inline test/extract.test.js]: @michael-ciniawsky stupid error :smile:

File: index.js [hunk 5]
```diff
@@ -122,7 +120,7 @@ function ExtractTextPlugin(options) {
 	if(isString(options)) {
 		options = { filename: options };
 	} else {
-		schemaTester(pluginSchema, options);
+		validateOptions('schema/plugin.json', options, 'Extract Text Plugin');
```
Comment (michael-ciniawsky) [inline index.js]: Just to double check `=> ./ <= schema...` ?  (validateOptions use `require()` 'internally')

File: index.js [hunk 6]
```diff
@@ -201,7 +199,7 @@ ExtractTextPlugin.prototype.extract = function(options) {
 	if(Array.isArray(options) || isString(options) || typeof options.options === "object" || typeof options.query === 'object') {
 		options = { loader: options };
 	} else {
-		schemaTester(loaderSchema, options);
+		validateOptions('schema/loader.json', options, 'Extract Text Plugin (Loader)');
```
Comment (michael-ciniawsky) [inline index.js]: Same as above 😛

**Pull request merged**